lua/telescope/utils.lua

```diff
@@ -116,10 +152,34 @@ utils.diagnostics_to_tbl = function(opts)
     for _, diag in pairs(diags) do
       -- workspace diagnostics may include empty tables for unused bufnr
       if not vim.tbl_isempty(diag) then
-        table.insert(items, preprocess_diag(diag, bufnr))
+        if filter_diag_severity(opts, diag.severity) then
+          table.insert(items, preprocess_diag(diag, bufnr))
+        end
       end
     end
   end
+
+  -- sort results by bufnr (prioritize cur buf), severity, lnum
+  table.sort(items, function(a, b)
+    if a.bufnr == b.bufnr then
+      if a.type == b.type then
+        return a.lnum < b.lnum
+      else
+        return a.type < b.type
+      end
+    else
+      -- prioritize for current bufnr
+      if a.bufnr == current_buf then
+        return true
+      end
+      if b.bufnr == current_buf then
+        return false
+      end
+      return a.bufnr < b.bufnr
+    end
+    end
+  )
```

Comment:
I think it should be indented this way

```suggestion
      return a.bufnr < b.bufnr
    end
  end)
```